pyvista/plotting/scalar_bars.py

```diff
@@ -28,12 +28,34 @@ def __repr__(self):
         """Nice representation of this class."""
         lines = []
         lines.append('Scalar Bar Title     Interactive')
-        for title, actor in self._scalar_bar_actors.items():
+        for title in self._scalar_bar_actors:
             interactive = title in self._scalar_bar_widgets
             title = f'"{title}"'
             lines.append(f'{title:20} {str(interactive):5}')
         return '\n'.join(lines)
 
+    def _remove_mapper_from_plotter(self, actor, reset_camera=False, render=False):
+        """Remove an actor's mapper from the given plotter's _scalar_bar_mappers."""
+        try:
+            mapper = actor.GetMapper()
+        except AttributeError:
+            return
+        for name in list(self._scalar_bar_mappers.keys()):
+            try:
+                self._scalar_bar_mappers[name].remove(mapper)
+            except ValueError:
+                pass
+            if len(self._scalar_bar_mappers[name]) < 1:
```

Comment:
If I'm reading this correctly we only need the `if` when the removal is successful on line 45. This would mean that we could replace `pass` on line `47` with `continue` to signal there's nothing to do for this key.

Also nitpick, but we're looking for an empty list so it would be idiomatic to do
```suggestion
            except ValueError:
                pass
            if not self._scalar_bar_mappers[name]:
```

Reply:
Sorry, I didn't add the `continue` to this suggestion, because I wasn't sure I was right. If you edit because of the dict size change in the loop you could add this too.

Reply:
I'm going to play it safe here.  If the mapper has somehow been removed, we should also remove the scalar bar, actors, slots, etc.  So, it will be `pass` instead of `continue`